### ISSUE TITLE
[Bash] Nested variable substitution doesn't highlight correctly

### ISSUE THREAD

**Issue by stevekuznetsov** [sublimehq/Packages#465]
In the following Bash code:

``` sh
        status="$( grep -Po "(?<=\-${id}\]).*" "${base_dir}/progress.log" | tail -n 1 )"
        status="${status#"${status%%[![:space:]]*}"}"

        if grep -q '• Failure' "${machine}/vagrant.log"; then
            id+="[$( grep --count '• Failure' "${machine}/vagrant.log" )f]"
        fi
```

We see the following highlighting:
![screenshot from 2016-06-08 08-14-05](https://cloud.githubusercontent.com/assets/7328370/15893954/fe48bf02-2d50-11e6-9afd-f9e10f32ce6a.png)

Syntax highlighting is incorrect in the inner variable expansion and also incorrect after the entire stanza.

The trim string after `%%` should not be categorized as a variable name in:

``` sh
"${status%%[![:space]]*}"
```

Also, while the above code snippet is valid, it is _also_ valid to remove the quotations around the inner expansion if you don't mind the breaking on whitespace behavior, which leaves us with:

``` sh
        status="$( grep -Po "(?<=\-${id}\]).*" "${base_dir}/progress.log" | tail -n 1 )"
        status="${status#${status%%[![:space:]]*}}"

        if grep -q '• Failure' "${machine}/vagrant.log"; then
            id+="[$( grep --count '• Failure' "${machine}/vagrant.log" )f]"
        fi
```

This highlights correctly for the most part -- the last brace is highlighted as a string, but the text _after_ is highlighted correctly.

![screenshot from 2016-06-08 08-21-14](https://cloud.githubusercontent.com/assets/7328370/15894107/0dc401ca-2d52-11e6-8dd7-8ac03847bd08.png)

**Comment by stevekuznetsov** [comment]
I'm Sublime 3 build 3114

**Comment by stevekuznetsov** [comment]
Also, is this the correct place to also report bugs with `subl` ? See:
```
$ subl --help
Sublime Text build 3114

Usage: sublime_text [arguments] [files]         edit the given files
   or: sublime_text [arguments] [directories]   open the given directories

Arguments:
  --project <project>: Load the given project
  --command <command>: Run the given command
  -n or --new-window:  Open a new window
  -a or --add:         Add folders to the current window
  -w or --wait:        Wait for the files to be closed before returning
  -b or --background:  Don't activate the application
  -h or --help:        Show help (this message) and exit
  -v or --version:     Show version and exit

Filenames may be given a :line or :line:column suffix to open at a specific
location.

$ subl --version
$ subl -v
$
```

The CLI doesn't seem to want to print the version information.

**Comment by keith-hall** [comment]
bugs relating to ST3 itself should be logged here: https://github.com/SublimeTextIssues/Core

**Comment by stevekuznetsov** [comment]
@keith-hall cheers, will report the CLI thing there

**Comment by wbond** [comment]
This bug will be addressed with #434

**Comment by stevekuznetsov** [comment]
@wbond always the bearer of good news!

**Comment by michaelblyons** [comment]
This is fixed, yes? #434 was closed with #1057

**Comment by FichteFoll** [comment]
Yes, looks fixed to me

![img](https://x0.at/5FZ.png)
